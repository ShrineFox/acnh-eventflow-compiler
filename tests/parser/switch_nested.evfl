flow Test():
    switch TestActor1.Query0():
        case 1:
            switch TestActor1.Query1(1):
                case 0:
                    TestActor0.Action0()
    TestActor0.Action1(0)
