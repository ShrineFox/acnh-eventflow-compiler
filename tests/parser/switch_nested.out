RootNode[name=Test, vardefs=[], out_edges=[Event6]]
ActionNode[name=Event0, action=EventFlowActionAction0(), params={}, out_edges=[Event7]]
SwitchNode[name=Event3, query=EventFlowQueryQuery1(param0), params={'param0': TypedValue(type=Type(type='int'), value=1)}, cases={'Event0': [0], 'Event7': [1]}, out_edges=[Event0, Event7]]
SwitchNode[name=Event6, query=EventFlowQueryQuery0(), params={}, cases={'Event3': [1], 'Event7': [0, 2]}, out_edges=[Event3, Event7]]
ActionNode[name=Event7, action=EventFlowActionAction1(param0), params={'param0': TypedValue(type=Type(type='int'), value=0)}, out_edges=[]]

